ClassCastException when combining OpenTelemetry with EventBus

### Describe the bug

When trying to send a message via a vertx EventBus, the call fails because OpenTelemetry tries to cast the message into an HttpRequest.

```
org.jboss.resteasy.spi.UnhandledException: java.lang.ClassCastException: class io.vertx.core.eventbus.impl.MessageImpl cannot be cast to class io.vertx.core.spi.observability.HttpRequest (io.vertx.core.eventbus.impl.MessageImpl and io.vertx.core.spi.observability.HttpRequest are in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @313ac989)
        at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:105)
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:359)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:91)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassCastException: class io.vertx.core.eventbus.impl.MessageImpl cannot be cast to class io.vertx.core.spi.observability.HttpRequest (io.vertx.core.eventbus.impl.MessageImpl and io.vertx.core.spi.observability.HttpRequest are in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @313ac989)
        at io.quarkus.opentelemetry.runtime.tracing.vertx.OpenTelemetryVertxTracer.sendRequest(OpenTelemetryVertxTracer.java:138)
        at io.quarkus.opentelemetry.runtime.tracing.vertx.OpenTelemetryVertxTracer.sendRequest(OpenTelemetryVertxTracer.java:42)
        at io.vertx.core.eventbus.impl.OutboundDeliveryContext.next(OutboundDeliveryContext.java:125)
        at io.vertx.core.eventbus.impl.EventBusImpl.sendOrPubInternal(EventBusImpl.java:394)
        at io.vertx.core.eventbus.impl.EventBusImpl.sendOrPubInternal(EventBusImpl.java:400)
        at io.vertx.core.eventbus.impl.EventBusImpl.send(EventBusImpl.java:103)
        at io.vertx.core.eventbus.impl.EventBusImpl.send(EventBusImpl.java:97)
        at io.vertx.mutiny.core.eventbus.EventBus.send(EventBus.java:94)
        at org.acme.opentelemetry.TracedResource.hello(TracedResource.java:20)
        at org.acme.opentelemetry.TracedResource_Subclass.hello$$superforward1(Unknown Source)
        at org.acme.opentelemetry.TracedResource_Subclass$$function$$1.apply(Unknown Source)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
        at io.quarkus.resteasy.runtime.QuarkusRestPathTemplateInterceptor.restMethodInvoke(QuarkusRestPathTemplateInterceptor.java:31)
        at io.quarkus.resteasy.runtime.QuarkusRestPathTemplateInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:50)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:62)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:49)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
        at org.acme.opentelemetry.TracedResource_Subclass.hello(Unknown Source)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
        at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
        at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:408)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:69)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
        ... 15 more
```


### Expected behavior

No exception occurs and an OpenTelemetry span is created.

### Actual behavior

Call fails with a ClassCastException, see above.

### How to Reproduce?

Create opentelemetry-quickstart project (see official quarkus guide for OpenTelemetry):

```
mvn io.quarkus.platform:quarkus-maven-plugin:2.6.1.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=opentelemetry-quickstart \
    -DclassName="org.acme.opentelemetry.TracedResource" \
    -Dpath="/hello" \
    -Dextensions="resteasy,quarkus-opentelemetry-exporter-otlp"
cd opentelemetry-quickstart
```

Then add the quarkus-vertx extension to the pom.xml (see official reactive event bus guide):

```
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-vertx</artifactId>
    </dependency>
```

Then expand the TracedResource class as follows:

```java
@Path("/hello")
public class TracedResource {
    @Inject
    EventBus eventBus;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        this.eventBus.send("bus", "Hello EventBus");
        return "Hello RESTEasy";
    }

    @ConsumeEvent("bus")
    public void onBusEvent(String msg) {
        System.out.println("Received " + msg);
    }
}
```

Start the application and call the endpoint. The exception is visible in the log.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "11.0.9" 2020-10-20 LTS

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)

### Additional information

_No response_

--- a/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/OpenTelemetryVertxTracer.java
+++ b/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/OpenTelemetryVertxTracer.java
@@ -81,6 +81,10 @@ public class OpenTelemetryVertxTracer
             final Iterable<Map.Entry<String, String>> headers,
             final TagExtractor<R> tagExtractor) {
 
+        if (!(request instanceof HttpRequest)) {
+            return null;
+        }
+
         io.opentelemetry.context.Context parentContext = context.getLocal(ACTIVE_CONTEXT);
         if (parentContext == null) {
             parentContext = io.opentelemetry.context.Context.root();
@@ -107,6 +111,10 @@ public class OpenTelemetryVertxTracer
             return;
         }
 
+        if (!(response instanceof HttpResponse)) {
+            return;
+        }
+
         Scope scope = spanOperation.getScope();
         if (scope == null) {
             return;
@@ -129,6 +137,10 @@ public class OpenTelemetryVertxTracer
             final String operation,
             final BiConsumer<String, String> headers,
             final TagExtractor<R> tagExtractor) {
+
+        if (!(request instanceof HttpRequest)) {
+            return null;
+        }
 
         io.opentelemetry.context.Context parentContext = context.getLocal(ACTIVE_CONTEXT);
         if (parentContext == null) {
@@ -154,6 +166,10 @@ public class OpenTelemetryVertxTracer
             final TagExtractor<R> tagExtractor) {
 
         if (spanOperation == null) {
+            return;
+        }
+
+        if (!(response instanceof HttpResponse)) {
             return;
         }
 

